runtime/vam/expr/agg/distinct.go

```diff
@@ -28,7 +28,7 @@ func (d *distinct) Consume(vec vector.Any) {
 		b.Truncate()
 		vec.Serialize(&b, i)
 		d.buf = binary.AppendVarint(d.buf[:0], int64(id))
-		d.buf = append(d.buf, b.Bytes().Body()...)
+		d.buf = scode.Append(d.buf, b.Bytes().Body())
```

Comment:
scode.Append adds the same bytes removed by scode.Bytes.Body.

```suggestion
		d.buf = append(d.buf, b.Bytes()...)
```